### PR TITLE
Update dependabot to ensure all projects are added

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -9,178 +9,238 @@ updates:
...
     schedule:
       interval: "weekly"
   - package-ecosystem: "gomod"
-    directory: "/extension/observer"
+    directory: "/exporter/stackdriverexporter"
     schedule:
       interval: "weekly"
+
+  ##################
+  #   PROCESSORS   #
```
Comment (jpkrohling) [inline .github/dependabot.yml:134]: It's missing the `/processor/groupbytraceprocessor` in this list.

**Review by jpkrohling — COMMENTED** [review]
One processor is missing. Here's a script I used to verify:

```bash
$ find . -name go.mod | sed "s~^\./~/~" | sed "s/\/go.mod$//" | sort > /tmp/gomods.txt
$ grep directory .github/dependabot.yml | awk '{print $2}' | sed 's/"//gi' | sort > /tmp/gomods-dependabot.txt
$ diff /tmp/gomods.txt /tmp/gomods-dependabot.txt 
```

The only other difference is about "/", which is a problem with a `sed` above.

File: .github/dependabot.yml (hunk 1 above)
Comment (bogdandrutu) [reply .github/dependabot.yml:134]: It is here, I added in the second commit last night

Comment (jpkrohling) [reply .github/dependabot.yml:134]: Indeed. The difference that I see is that this PR isn't based on the current master and lacks the processor itself. 

```console
$ find . -name go.mod | sed "s~^\./~/~" | sed "s/\/go.mod$//" | sort > /tmp/gomods.txt
$ grep directory .github/dependabot.yml | awk '{print $2}' | sed 's/"//gi' | sort > /tmp/gomods-dependabot.txt
$ diff /tmp/gomods.txt /tmp/gomods-dependabot.txt 
1c1
< 
---
> /
28a29
> /processor/groupbytraceprocessor

$ ls -1 processor/
k8sprocessor
metricstransformprocessor
resourcedetectionprocessor
routingprocessor

```

**Review by jpkrohling — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (bogdandrutu) [reply .github/dependabot.yml:134]: Nice :)

**Pull request merged**